github_connector_odoo/models/github_repository_branch.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2016-Today: Odoo Community Association (OCA)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+import os
+
+from os.path import join as opj
+
+from openerp import models, fields, api
+
+from openerp.modules import load_information_from_description_file
+
+# Hard define this value to make this module working with or without
+# the patch (that backports V10 manifests analysis code.
+MANIFEST_NAMES = ('__manifest__.py', '__openerp__.py')
+
+_logger = logging.getLogger(__name__)
+
+
+class GithubRepositoryBranch(models.Model):
+    _inherit = ['github.repository.branch']
+
+    module_paths = fields.Text(
+        string='Module Paths', help="Set here extra relative paths"
+        " you want to scan to find modules. If not set, root path will be"
+        " scanned. One repository per line. Example:\n"
+        "./addons/\n"
+        "./openerp/addons/")
+
+    module_version_ids = fields.One2many(
+        comodel_name='odoo.module.version',
+        inverse_name='repository_branch_id', string='Module Versions')
+
+    module_version_qty = fields.Integer(
+        string='Number of Module Versions',
+        compute='_compute_module_version_qty')
+
+    runbot_url = fields.Char(
+        string='Runbot URL', compute='_compute_runbot_url')
+
+    # Compute Section
+    @api.multi
+    @api.depends(
+        'name', 'repository_id.runbot_id_external',
+        'organization_id.runbot_url_pattern')
+    def _compute_runbot_url(self):
+        for branch in self:
+            branch.runbot_url =\
+                branch.organization_id.runbot_url_pattern.format(
```

Comment:
Add some robustness also here, in case _runbot_url_pattern_ does not contain values.